feat: priority fee calculation strategies

spypsy opened the pull request:
This PR adds an L1 priority fee analysis tool for nodes running in fisherman mode.

Instead of clearing all queued requests from the publisher, fisherman nodes will now calculate a priority fee given a specific strategy, then wait for the ethereum block to be mined and make an analysis on how competitive the fee we calculated was.

This can be done with multiple strategies.

Analysis is logged & pushed to our metrics on each strategy as attribute

Fixes [A-307](https://linear.app/aztec-labs/issue/A-307/observability-and-metrics-on-l1-gas-fee-calculation-in-fisherman-mode)

Used Claude a lot for the analysis parts since they don't affect production node code paths.
Not for the update on L1 tx utils that now uses a strategy object for calculations (making it easy to swap. could possibly be made into a config value)

PhilWindle commented on yarn-project/ethereum/src/l1_tx_utils/l1_fee_analyzer.ts:
```diff
@@ -0,0 +1,875 @@
+
+    try {
+      // Poll for the next block (with timeout)
+      const maxWaitMs = 13_000; // 13 seconds max wait
```
Why don't we use `retryUntil` here?

PhilWindle commented on yarn-project/ethereum/src/l1_tx_utils/l1_fee_analyzer.ts:
```diff
@@ -0,0 +1,875 @@
+          continue;
+        }
+
+        const priorityFee = tx.maxPriorityFeePerGas || 0n;
```
I'm sure I've seen this block of code duplicated at least once before.

PhilWindle reviewed (changes requested):
There is a lot of AI generated code here. I understand the reasoning why but have concerns around a couple of areas.

1. There are no tests
2. We use `Promise<unknown>` and `as any` in a number of places
3. There are places that could do with a refactor to identify code used in multiple places.

spypsy committed:
feat: priority fee calculation strategies

This PR adds an L1 priority fee analysis tool for nodes running in fisherman mode.

Instead of clearing all queued requests from the publisher, fisherman nodes will now calculate a priority fee given a specific strategy, then wait for the ethereum block to be mined and make an analysis on how competitive the fee we calculated was.

This can be done with multiple strategies.

Analysis is logged & pushed to our metrics on each strategy as attribute

Fixes [A-307](https://linear.app/aztec-labs/issue/A-307/observability-and-metrics-on-l1-gas-fee-calculation-in-fisherman-mode)

Used Claude a lot for the analysis parts since they don't affect production node code paths.
Not for the update on L1 tx utils that now uses a strategy object for calculations (making it easy to swap. could possibly be made into a config value)

spypsy committed:
update dashboard

spypsy committed:
Fix namespace selector

spypsy committed:
Track gas spent by each blob tx in a block
